src/ape/api/providers.py

```diff
@@ -598,6 +598,76 @@ def get_transaction_trace(  # type: ignore[empty-body]
             Iterator(:class:`~ape.type.trace.TraceFrame`): Transaction execution trace.
         """
 
+    @raises_not_implemented
+    def poll_blocks(  # type: ignore[empty-body]
+        self,
+        stop_block: Optional[int] = None,
+        required_confirmations: Optional[int] = None,
+        new_block_timeout: Optional[int] = None,
+    ) -> Iterator[BlockAPI]:
+        """
+        Poll new blocks.
+
+        **NOTE**: When a chain reorganization occurs, this method logs an error and
+        yields the missed blocks, even if they were previously yielded with different
+        block numbers.
+
+        **NOTE**: This is a daemon method; it does not terminate unless an exception occurs
+        or a ``stop`` is given.
+
+        Args:
+            start_block (Optional[int]): The block number to start with. Defaults to the pending
+              block number.
+            stop_block (Optional[int]): Optionally set a future block number to stop at.
```

Comment:
**note for myself as the reviewer**: so the doc here makes it sound like it's inclusive, meaning the you enter the last block you want here and not the end_range value. This is how `poll_blocks` worked before because only the method named `range()` behave like the python builtin `range()` method